### PR TITLE
Add a basic library to query the Nexus API

### PR DESCRIPTION
**Pull request by olive42**
This will be later used first as a LWRP to provide an easy way to list, create
or update repositories.

### CODE REVIEW

File: libraries/api.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+module Nexus3
+  class Api
+    require 'json'
+
+    def initialize(url, user, password)
+      @http_client = HTTPClient.new(url).tap do |client|
```
Comment (Annih) [inline libraries/api.rb]: you should pass your url using the `base_url` name parameter (https://github.com/nahi/httpclient/blob/master/lib/httpclient.rb#L398)

File: libraries/api.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        # Authentication
+        client.set_auth(nil, user, password)
+        # Debugging
+        client.debug_dev = STDOUT if ::Chef::Log.debug? || (ENV['HTTPCLIENT_LOG'] == 'stdout')
+      end
+      @url = url
```
Comment (Annih) [inline libraries/api.rb]: I think you don't need to store the url as an instance_variable

File: spec/unit/libraries/api_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    ]
+  end
+
+  before(:each) do
+    WebMock.disable_net_connect!
+  end
+
+  describe 'list_repositories' do
+    it 'returns repos' do
+      # TODO: if we add the Authorization header, the test fails, so
```
Comment (Annih) [inline spec/unit/libraries/api_spec.rb]: HTTPClient send basic auth header only after a 401 error.
If you want to always send the authentication header use the force_basic_auth property https://github.com/nahi/httpclient/blob/master/lib/httpclient.rb#L592

File: libraries/api.rb (hunk 1 above)
Comment (olive42) [inline libraries/api.rb]: Done

File: libraries/api.rb (hunk 2 above)
Comment (olive42) [inline libraries/api.rb]: Removed

File: spec/unit/libraries/api_spec.rb (hunk 3 above)
Comment (olive42) [inline spec/unit/libraries/api_spec.rb]: Fixed.

File: libraries/api.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+module Nexus3
+  # Interact with the Nexus3 API
+  class Api
+    require 'json'
+
+    def initialize(base_url, user, password)
+      @http_client = HTTPClient.new(base_url: base_url).tap do |client|
+        # Authentication
+        client.set_auth(base_url, user, password)
+        client.force_basic_auth = true
```
Comment (Annih) [inline libraries/api.rb]: OPT: could be pass as constructor parameter

**Pull request merged**